### DISCUSSION TITLE
Enhancement: Crafting table affects crafting recipe.

### DISCUSSION THREAD

**Discussion by PyroTech03** [AscensionGameDev/Intersect-Engine#2228] [Ideas]
Currently we define crafts in the craft editor for resources needed and failure and consumption on failure. We build a crafting table to allow access to the various crafts and that's it.

If we wanted to make a table that was more or less effecient, we'd have to make alternatives of each craft to import into the new table, doubling the amount of crafts for each effeciency we want to implement.

I'm suggesting we add a property to crafting tables that we can define a number, either static or based on a variable, that would affect the crafting recipe by that amount to a minimum of at least needing 1 of an item in the recipe. 

In the case of say a quiver takes 10 leather and 1 string to make. If you're on a table that has a 20% effeciency set on it, that recipe would now cost 8 leather and 1 string. If we say it had a base failure chance of 75% we now say it has a 55% failure. 

This allows much cleaner setup by only having to define a base recipe. And by allowing the number to be tied to a variable it could be used to allow crafting skills to actually effect the craft or something dynamic like if we're doing any kind of environment or economy simulation.

**Comment by EspectroW94** [comment]
I would like to complement this idea, it would also be good to be able to vary the probability of failure in some way, whether due to variables or luck affecting it, the luckier the character is, the lower the probability of failure.

**Reply by PyroTech03** [reply]
From my understanding, luck affects anything with a percent value already. I have not tested this but it's what I've been told.

**Reply by EspectroW94** [reply]
I created some items with 100% luck, by equipping them and crafting something with 60% failure chance. It keeps failing the same way.

**Reply by PyroTech03** [reply]
Good to know I guess.
